File: Processes/Python/fork_example.py
import os
import sys

pid = os.fork()
if pid < 0:
    print("fork failed", file=sys.stderr)
    sys.exit(1)
elif pid == 0:
    print(f"Child process: PID = {os.getpid()}, Parent PID = {os.getppid()}")
else:
    print(f"Parent process: PID = {os.getpid()}, Child PID = {pid}")
    os.wait()
    print("Child finished")

File: Processes/C/fork_example.c
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <stdio.h>
#include <stdlib.h>

int main() {
    pid_t pid = fork();
    if (pid < 0) {
        perror("fork failed");
        exit(1);
    } else if (pid == 0) {
        printf("Child process: PID = %d, Parent PID = %d\n", getpid(), getppid());
    } else {
        printf("Parent process: PID = %d, Child PID = %d\n", getpid(), pid);
        wait(NULL);
        printf("Child finished\n");
    }
    return 0;
}